src/text_formatter/text_formatter.ts

```diff
@@ -57,42 +47,39 @@ export function textToListFormat(input: String) {
  * @return a string-array where each element is one word
  */
 export function sentenceIntoWordList(input: String) {
-	const separator = /[ ,]+/; // List all the values where the content of a sentence will get splitted: whitespace[ ] and comma[,]
-	const splittedSentenceIntoListOfWords = input.split(separator);
-	const temp = [];
+
+	const splittedSentenceIntoListOfWords = input.split(wordSeparator);
+	const listOfWords = [];
 	for (let index = 0; index < splittedSentenceIntoListOfWords.length; index++) {
-		// remove all elements(senteces) which are a empty String after splitting
-		if (splittedSentenceIntoListOfWords[index].trim() == "") {
+		if (splittedSentenceIntoListOfWords[index].trim() == "") {	// remove all elements(senteces) which are a empty String after splitting
 			splittedSentenceIntoListOfWords.splice(index, 1);
```

Comment:
Why do you remove these "bad" elements from `splittedSentenceIntoListOfWords`? You're putting the "good" ones into `listOfWords` already. And you don't need `splittedSentenceIntoListOfWords` anymore.

Reply:
And btw, repeated splicing is just inefficient, since small lists in JS will be ArrayLists.